src/main/java/it/gov/pagopa/payhub/auth/service/a2a/registration/ClientSecretGeneratorService.java

```diff
@@ -0,0 +1,22 @@
+package it.gov.pagopa.payhub.auth.service.a2a.registration;
+
+import it.gov.pagopa.payhub.auth.service.DataCipherService;
+import org.springframework.stereotype.Service;
+
+import java.util.Base64;
+import java.util.function.Function;
+
+@Service
+public class ClientSecretGeneratorService implements Function<String, String> {
+
+	private final DataCipherService dataCipherService;
+
+	public ClientSecretGeneratorService(DataCipherService dataCipherService) {
+		this.dataCipherService = dataCipherService;
+	}
+
+	@Override
+	public String apply(String clientSecret) {
```

Comment:
A class named ClientSecretGeneratorService is supposed to generate the `clientSecret`, not to have it as input argument